### DISCUSSION TITLE
Unbounded Bin Problem

### DISCUSSION THREAD

**Discussion by laukik-target** [google/or-tools#3756] [CP-SAT questions]
**Title:** 
Non-finite Consumables Optimization in Packing Function

**Issue:**

I am using the `from ortools.sat.python import cp_model` in my Python code to optimize packing of items into consumables (containers). However, I am facing an issue where the consumables are not being reused efficiently, leading to unused volume and less space-efficient packing.

**Code:**

I have a function `optimize_packing(items_data, consumables)` which takes in two parameters:

- `items_data`: A list of dictionaries, each representing an item with properties like SKU, volume, weight, etc.
- `consumables`: A list of dictionaries, each representing a consumable with properties like width, height, length, etc.

The function uses the CP-SAT solver to assign items to consumables, ensuring that the total volume and weight of items in each consumable do not exceed its capacity and a maximum weight limit of 25kgs, respectively. The goal is to maximize the total weight of packed items.

**Problem:**

The current output of the function is not as expected. The function is assigning each item to a new consumable, even when there is enough space in a previously used consumable. This leads to inefficient use of consumables and more unused volume.

**Example:**

For example, given two identical items and a list of consumables, the function assigns each item to a different consumable, even though both items could fit into one consumable.

**Expected Behavior:**

The function should assign both items to the same consumable, as there is enough space for both. This would lead to more efficient use of consumables and less unused volume.

**Request:**

I am not sure how to modify the function to achieve the expected behavior. I believe it might be related to the concept of non-finite or unbounded consumables, where a consumable can be reused if there is enough space left after packing previous items. I would appreciate any guidance or suggestions on how to solve this issue.

**Additional Context:**

Please find the detailed code, input, and output examples in the attached files.

**Comment by lperron** [comment]
Please be precise. 

_Describe the solution you'd like
not sure yet_

is not very informative. What help do you expect to receive ?

**Reply by laukik-target** [reply]
> Please be precise.
> 
> _Describe the solution you'd like not sure yet_
> 
> is not very informative. What help do you expect to receive?


Hey I have updated by issue, hope its informative now
Let me know if you still need any info
